blockchain/Validation.go

```diff
@@ -3095,6 +3103,54 @@ func LoadBlockIndex(params *msg.BitcoinParams) bool {
 	return true
 }
 
+func InitBlockIndex(param *msg.BitcoinParams) (ret bool) {
+	// todo:LOCK(cs_main);
+
+	// Check whether we're already initialized
+	if GChainActive.Genesis() != nil {
+		return true
+	}
+
+	// Use the provided setting for -txindex in the new database
+	GfTxIndex = utils.GetBoolArg("-txindex", DEFAULT_TXINDEX)
+	// todo:pblocktree->WriteFlag("txindex", fTxIndex)
+	logger.GetLogger().Info("Initializing databases...")
+
+	// Only add the genesis block if not reindexing (in which case we reuse the
+	// one already on disk)
+	if !GfReindex {
+		ret = true
+		defer func() {
+			if err := recover(); err != nil {
+				logger.ErrorLog(fmt.Sprintf("LoadBlockIndex(): failed to initialize block database: %s", err))
+				ret = false
+			}
+		}()
+
+		block := param.GenesisBlock.Block
+		// Start new block file
+		nBlockSize := block.SerializeSize()
+		var (
+			blockPos *model.DiskBlockPos
```

Comment:
not define pointer model.DiskBlockPos; 
because you call FindBlockPos, this function direct assignment to the pointer of element, this could cause panic when the pointer is nil.